include/wlr/backend.h

```diff
@@ -1,7 +1,29 @@
 #ifndef _WLR_BACKEND_H
 #define _WLR_BACKEND_H
 
-struct wlr_backend *wlr_backend_init();
-void wlr_backend_free(struct wlr_backend *backend);
+#include <wayland-server.h>
+
+struct wlr_backend_impl;
+struct wlr_backend_state;
+
+struct wlr_backend {
+	const struct wlr_backend_impl *impl;
+	struct wlr_backend_state *state;
```

Comment:
This is a DRM-specific type. Perhaps this should be changed to a void *, or we could have 'wlr_backend' as a member of 'wlr_\*_backend' and use wl_container_of() to get to a backend-specific structure (the same way the session code currently does it).